tsdb/engine/tsm1/file_store.go

```diff
@@ -924,7 +924,9 @@ func (f *FileStore) BlockCount(path string, idx int) int {
 				}
 			}
 			_, _, _, _, _, block, _ := iter.Read()
-			return BlockCount(block)
+			// on Error, BlockCount(block) returns 0 for cnt
+			cnt, _ := BlockCount(block)
```

Comment:
Missing the error handler here

Reply:
this function returns `0` for other edge cases, the caller should expect the `0` as failure. 
I think we should collect the errors somewhere but not everywhere !!